File: src/main/java/itmo/is/project/dto/trade/TradeRequest.java
package itmo.is.project.dto.trade;

import com.fasterxml.jackson.annotation.JsonProperty;
import itmo.is.project.dto.resource.ResourceIdAmountDto;

import java.util.List;

public record TradeRequest(

        @JsonProperty(value = "buy", required = true)
        List<ResourceIdAmountDto> buy,

        @JsonProperty(value = "sell", required = true)
        List<ResourceIdAmountDto> sell
) {
}

File: src/main/kotlin/itmo/isproject/dto/trade/TradeRequest.kt
package itmo.isproject.dto.trade

import com.fasterxml.jackson.annotation.JsonProperty
import itmo.isproject.dto.resource.ResourceIdAmountDto

data class TradeRequest(

    @JsonProperty(value = "buy", required = true)
    val buy: List<ResourceIdAmountDto>,

    @JsonProperty(value = "sell", required = true)
    val sell: List<ResourceIdAmountDto>
)
